src/components/FileTreeNode/index.tsx

```diff
@@ -296,7 +304,7 @@ export class FileTreeNodeBase<TreeNodeType> extends React.Component<Props> {
         validationURL={version.validationURL}
         selectedPath={version.selectedPath}
       >
-        {this.renderWithLinterInfo}
+        {(info: LinterProviderInfo) => this.renderWithLinterInfo(info)}
```

Comment:
Could you give me some steps to reproduce the stale rendering state that led you to add this? I would guess that the entry status data wasn't changing, but how do you enter that state?

I would like to keep this as `{this.renderWithLinterInfo}`, if we can, because otherwise the anonymous function could trigger constant re-renders of this component. It might not be a problem but I tried testing it out and ran into ﻿https://github.com/mozilla/addons-code-manager/issues/1068 (which is unrelated to your patch). Then I fell into a rabbit hole, heh. I will continue reviewing your patch tomorrow.

Reply:
When using `{this.renderWithLinterInfo}` in this patch,
- Go to http://localhost:3000/en-US/compare/502955/versions/1541794...1541798/
- Change old version from 1.4.0 to 1.3.0
The entry status style (`wasModified`) for `META-INF` and `manifest.json` are not shown. Then clicking any entry will cause `FileTreeNode` to re-render and show entry status.